packages/block-library/src/media-text/index.js

```diff
@@ -82,6 +83,63 @@ export const settings = {
 		align: [ 'wide', 'full' ],
 	},
 
+	transforms: {
+		from: [
+			{
+				type: 'block',
+				blocks: [ 'core/image' ],
+				transform: ( { alt, url, id } ) => (
+					createBlock( 'core/media-text', {
+						mediaAlt: alt,
+						mediaId: id,
+						mediaUrl: url,
+						mediaType: 'image',
+					} )
+				),
+			},
+			{
+				type: 'block',
+				blocks: [ 'core/video' ],
+				transform: ( { src, id } ) => (
+					createBlock( 'core/media-text', {
+						mediaId: id,
+						mediaUrl: src,
+						mediaType: 'video',
+					} )
+				),
+			},
+		],
+		to: [
+			{
+				type: 'block',
+				blocks: [ 'core/image' ],
+				isMatch: ( { mediaType, mediaUrl } ) => {
+					return ! mediaUrl || mediaType === 'image';
+				},
+				transform: ( { mediaAlt, mediaId, mediaUrl } ) => {
+					return createBlock( 'core/image', {
+						alt: mediaAlt,
+						id: mediaId,
+						url: mediaUrl,
+					} );
```

Comment:
What about the text of the inner blocks? Should we just return them in addition to this block to keep them after the image block?

Reply:
I thought about it, but unfortunately, our transforms API does not pass the child blocks. I think it would be a nice improvement in the API and I can work on it. At the time I did not add this because I was not sure if it collided with the freeze. I think we may manage to get something working without breaking any back compatibility I can give it a try.

Reply:
I think this is fine. This change, to me, is mostly about the ergonomics of quickly switching between "image-type" blocks. In the future, a more comprehensive transform will be a welcome addition; until then, the Undo button is what allows the user to try a conversion and roll back at any time.